alf/environments/wrappers.py

```diff
@@ -170,42 +175,37 @@ def __init__(self,
             shape=(self.screen_size, self.screen_size, 1),
             dtype=np.uint8)
 
-        self.game_over = True
-        self.lives = 0  # Will need to be set by reset().
+        self._lives = 0
 
     def reset_with_random_noops(self):
         self.env.reset()
+        self._lives = self.env.ale.lives()
         n_noops = self.env.unwrapped.np_random.randint(
             1, self.noop_max + 1)
         for _ in range(n_noops):
             _, _, game_over, _ = self.env.step(0)
             if game_over:
                 self.env.reset()
 
-    def fire_on_reset(self):
+    def fire(self):
         # The following code is from https://github.com/openai/gym/...
         # ...blob/master/gym/wrappers/atari_preprocessing.py
         action_meanings = self.env.unwrapped.get_action_meanings()
         if action_meanings[1] == 'FIRE' and len(action_meanings) >= 3:
             self.env.step(1)
             self.env.step(2)
 
-    def reset(self):
+    def reset(self, hard_reset=True):
```

Comment:
comment hard_reset. But instead of adding 'hard_reset', which totally changes the semantic of 'reset' for gym.Env, I'd suggest keeping reset same signature, but adding another function such as start_new_life() which is internally used in this class.

Reply:
added a start_new_life() function